### ISSUE TITLE
Operator with namespace-only permissions hangs during reconciliation

### ISSUE THREAD

**Issue by jpkrohling** [jaegertracing/jaeger-operator#935]
Split from #931, caused by operator-framework/operator-sdk#2608.

When the operator doesn't have cluster-scope permissions, it hangs during the reconciliation with the following message happening every second:

```
E0228 10:28:58.020049       1 reflector.go:123] pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/cache/reflector.go:96: Failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:observability:jaeger-operator" cannot list resource "secrets" in API group "" at the cluster scope
```

**Comment by ewohltman** [comment]
Unless I set my Jaeger Operator up incorrectly, this seems to be an issue even when intended with cluster-scope permissions:

```
E0301 02:01:03.884054       1 reflector.go:123] pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/cache/reflector.go:96: Failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:observability:jaeger-operator" cannot list resource "secrets" in API group "" at the cluster scope
```

Looking at the cluster role spec here, I don't see where permissions are given for secrets (if its supposed to): https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/cluster_role.yaml

**Comment by jpkrohling** [comment]
@ewohltman right, you are facing the problem that is being fixed as part of #936. This issue here is being addressed by #937: the lack of permissions will cause errors while reconciling the secrets, except that it won't block indefinitely.

**Comment by ewohltman** [comment]
> @ewohltman right, you are facing the problem that is being fixed as part of #936. This issue here is being addressed by #937: the lack of permissions will cause errors while reconciling the secrets, except that it won't block indefinitely.

@jpkrohling Thanks for pointing me in the right direction!
